zcm/julia/ZCM.jl

```diff
@@ -0,0 +1,334 @@
+module ZCM
+
+# RRR / Note: Julia requires that the memory layout of the C structs is consistent
+#             between their definitions in zcm headers and this file
+
+# C ptr types for types that we don't need the internals of
+module Native
+
+type Zcm
+end
+
+type Sub
+end
+
+type UvSub
+end
+
+type RecvBuf
+    recv_utime ::Int64;
+    zcm        ::Ptr{Native.Zcm};
+    data       ::Ptr{UInt8};
+    data_size  ::UInt32;
+end
+
+type EventLog
+end
+
+type EventLogEvent
+    eventnum  ::Int64;
+    timestamp ::Int64;
+    chanlen   ::Int32;
+    datalen   ::Int32;
+    channel   ::Ptr{UInt8};
+    data      ::Ptr{UInt8};
+end
+
+end
+
+# RRR: make variable naming consistent
+# RRR: improve orgainzation and make header comments more visible
+# Exported Objects and Methods
+type RecvBuf
+    rbuf  ::Ptr{Native.RecvBuf};
+    utime ::Int64;
+    data  ::Array{UInt8};
+
+    function RecvBuf(rbuf::Ptr{Native.RecvBuf})
+        instance  = new();
+        instance.rbuf = rbuf;
+
+        if (rbuf != C_NULL)
+            loadedRbuf = unsafe_load(rbuf);
+
+            instance.utime = loadedRbuf.recv_utime;
+
+            if (loadedRbuf.data != C_NULL)
+                instance.data = unsafe_wrap(Array, loadedRbuf.data, loadedRbuf.data_size);
+            else
+                instance.data = [];
+            end
+        end
+
+        # destruction of the underlying Native.RecvBuf pointer is handled by the C lib internally
+
+        return instance;
+    end
+end
+export RecvBuf;
+
+type __zcm_handler
+    handler::Any;
+    usr::Any;
```

Comment:
Instead of having `handler` and `usr` be of type `Any`, perhaps it makes more sense to parameterize `__zcm_handler` on the types of `handler` and `usr`, i.e.:
```julia
type __zcm_handler{H, U}
    handler::H
    usr::U
end
```
`Sub` would then also be parameterized on the concrete `__zcm_handler` subtype.

Reply:
Agreed, but this type may go away if I port in the code from LCMCore